myapp/spec/system/tasks_spec.rb

```diff
@@ -1,27 +1,47 @@
 require 'rails_helper'
 
+def login
+  visit '/login'
+  fill_in 'session[username]', with: 'user'
+  fill_in 'session[password]', with: 'test'
+  click_on 'Login'
+end
+
+def create_user
+  create(:user, username: 'user', password: 'test')
+end
+
 RSpec.describe 'Tasks', type: :system do
+
   describe 'page rendering' do
+    before do
+      create_user
+      login
+    end
+
+    let!(:user) { User.last }
```

Comment:
I think if there is no reason to get last user, looks good to me as follows
```suggestion
    let!(:user) { create(:user) }
```
This is because it is unpredictable which user will be the last, depending on the state of the database and the order of testing.
Tests should be unaffected by the state of the database or the order of the tests and should have consistent results.